lms/product/moodle/_plugin/grouping.py

```diff
@@ -34,7 +34,11 @@ def get_groups_for_learner(self, _svc, course, group_set_id):
             ):
                 return learner_groups
         except ExternalRequestError as exc:
-            if exc.status_code == 404:
+            if (
+                exc.validation_errors
+                # There are no error codes in Moodle's APIs
+                and exc.validation_errors.get("errorcode") == "invalidrecord"
```

Comment:
I think I get your point, but it's a bit counter intuitive to have a comment saying "There are no error codes" followed by an instruction reading something called "errorcode" 😅 

Perhaps the comment should say something in the lines of "Moodle's APIs don't return semantic HTTP status codes"

Reply:
Just saw this comment https://github.com/hypothesis/lms/pull/6105/files#diff-8ca2bbb29adf1f5d477eb4cb029000a47f0c2f85aeed0741b50dae180f38ca21R260-R261

That's more in the lines of my suggestion.